server/datastore/mysql/policies.go

```diff
@@ -690,3 +690,128 @@ func (ds *Datastore) CleanupPolicyMembership(ctx context.Context, now time.Time)
 
 	return nil
 }
+
+func (ds *Datastore) IncrementPolicyViolationDays(ctx context.Context, now time.Time) error {
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		return incrementViolationDaysDB(ctx, tx, now)
+	})
+}
+
+func incrementViolationDaysDB(ctx context.Context, tx sqlx.ExtContext, now time.Time) error {
+	const (
+		statsID        = 0
+		statsType      = "policy_violation_days"
+		updateInterval = 24 * time.Hour
+	)
+
+	// get current count of policy violation days from `aggregated_stats``
+	selectStmt := `
+		SELECT 
+			json_value, 
+			created_at, 
+			updated_at 
+		FROM 
+			aggregated_stats 
+		WHERE 
+			id = ? AND type = ?`
+	selectResult := []struct {
+		CreatedAt time.Time `json:"created_at" db:"created_at"`
+		UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
+		Count     uint      `json:"json_value" db:"json_value"`
+	}{}
+	if err := sqlx.SelectContext(ctx, tx, &selectResult, selectStmt, statsID, statsType); err != nil {
```

Comment:
Have you considered using `sqlx.GetContext` since you're expecting one or zero results from this?

Reply:
@chiiph, I did try that but I kept getting the following error and wasn't able to sort it out. Any tricks that I'm missing?

`scannable dest type slice with >1 columns (3) in result`

Reply:
oh wait, I think I just figured out what I was doing wrong 🤦‍♀️